lib/srv/db/mongodb/protocol/message.go

```diff
@@ -102,28 +102,44 @@ func readHeaderAndPayload(reader io.Reader) (*MessageHeader, []byte, error) {
 		return nil, nil, trace.BadParameter("invalid header size %v", header)
 	}
 
-	// Max BSON document size is 16MB
-	// https://www.mongodb.com/docs/manual/reference/limits/#mongodb-limit-BSON-Document-Size
-	if length-headerSizeBytes >= 16*1024*1024 {
-		return nil, nil, trace.BadParameter("exceeded the maximum document size, got length: %d", length)
+	payloadLength := int64(length - headerSizeBytes)
+	// TODO: get the max limit from Mongo handshake.
```

Comment:
Can you create an issue and add the link here, so we won't forget about it (or link existing one if exists)